packages/common/observability/src/observability.ts

```diff
@@ -0,0 +1,434 @@
+        this.gauge('dxos.client.space.epoch', sp.epoch, { key: sp.key });
+        this.gauge('dxos.client.space.currentDataMutations', sp.currentDataMutations, { key: sp.key });
+      }
+    });
+
+    const subscribeToSpaceUpdate = (space: Space) =>
+      space.pipeline.subscribe({
+        next: () => {
+          updateSpaceMetrics.emit();
+        },
+      });
+
+    spaces.forEach((space) => {
+      subscriptions.set(space.key.toHex(), subscribeToSpaceUpdate(space));
+    });
+
+    client.spaces.subscribe({
+      next: async () => {
+        spaces = client.spaces.get();
+        // spaces = await this.getSpaces(this._agent.client);
+        spaces
+          .filter((space) => !subscriptions.has(space.key.toHex()))
+          .forEach((space) => {
+            subscriptions.set(space.key.toHex(), subscribeToSpaceUpdate(space));
+          });
+      },
+    });
+
+    scheduleTaskInterval(this._ctx, async () => updateSpaceMetrics.emit(), DATADOG_IDLE_INTERVAL);
+  }
+
+  public async startRuntimeMetrics(client: Client, frequency: number = DATADOG_IDLE_INTERVAL) {
+    const platform = await client.services.services.SystemService?.getPlatform();
+    invariant(platform, 'platform is required');
+
+    this.setTag('platform_type', Platform.PLATFORM_TYPE[platform.type as number].toLowerCase());
+    if (this._mode === 'full') {
+      // platform[foo] does not work?
+      if (platform.platform) {
+        this.setTag('platform', platform.platform);
+      }
+      if (platform.arch) {
+        this.setTag('arch', platform.arch);
+      }
+      if (platform.runtime) {
+        this.setTag('runtime', platform.runtime);
+      }
+    }
+    scheduleTaskInterval(
+      this._ctx,
+      async () => {
+        log('platform');
+        client.services.services.SystemService?.getPlatform()
+          .then((platform) => {
+            log('platform', { platform });
+            if (platform.memory) {
+              this.gauge('dxos.client.runtime.rss', platform.memory.rss);
+              this.gauge('dxos.client.runtime.heapTotal', platform.memory.heapTotal);
+              this.gauge('dxos.client.runtime.heapUsed', platform.memory.heapUsed);
+            }
+          })
+          .catch((error) => log('platform error', { error }));
+      },
+      frequency,
+    );
+  }
+
+  /*********
+   Telemetry
+   *********/
+
+  public initTelemetry(batchSize = 30) {
+    if (this.config.TELEMETRY_API_KEY && this._mode !== 'disabled') {
+      this._segmentTelemetry = new SegmentTelemetry({
+        apiKey: this.config.TELEMETRY_API_KEY,
+        batchSize,
+        getTags: () => this._tags,
+      });
+      this._telemetryEnabled = true;
+    } else {
+      log('segment disabled');
+    }
+  }
+
+  get telemetryEnabled() {
+    return this._telemetryEnabled;
+  }
+
+  public telemetryEvent(eo: EventOptions) {
+    this._segmentTelemetry?.event(eo);
+  }
+
+  public telemetryPage(po: PageOptions) {
+    this._segmentTelemetry?.page(po);
+  }
```

Comment:
consider exposing `this._segmentTelemetry` as `telemetry` to enable api that looks like `obs.telemetry.page` instead of `obs.telemetryPage`

Reply:
agree that it looks less ugly but was trying to avoid coupling the provider to the capability (e.g. if we wanted to use something other than Segment or Prometheus instead of Datadog).

And also make it more obvious that the package will do things like add tags and change other behavior depending on the configuration.

Reply:
> trying to avoid coupling the provider to the capability

yeah, for sure, they're certainly a little coupled right now but the idea of `@dxos/telemetry` package initially was that it should provide an interface of which the underlying telemetry implementation could be swapped out. I think `obs.telemetry` could have a type of that interface and you wouldn't need to know about what it's hooked up to. The way it is now it feels like the `Observability` class has too much knowledge about the internals of these other classes in order to re-expose their methods.

Reply:
my intent was to try and group telemetry, metrics and error log collection into a single interface to make it easier to config, share tags such as identity/device keys.

I was the least familiar with the Telemetry and what was Segment-specific, so right now it's just passing through to the Segment API. Was thinking that the API surface of all of telemetry, metrics, and log collection would be small enough to avoid extra classes, but can see the benefit of the ergonomics of the API.